src/util/files.ts

```diff
@@ -0,0 +1,55 @@
+import { FileInfo } from '../global/shared-types/file.types';
+import { createRandomString } from './random-string';
+import {
+    getFileBackgroundColor,
+    getFileColor,
+    getFileExtensionTitle,
+    getFileIcon,
+} from './file-metadata';
+
+export function createFileInfo(file: File): FileInfo {
+    let limeFile: FileInfo = {
+        id: createRandomString(),
+        filename: file.name,
+        contentType: file.type,
+        size: file.size,
+        fileContent: file,
+    };
+    limeFile = {
```

Comment:
I suppose this is one way, but I think we can just do `limeFile.icon = ...`

Reply:
I like your way more. See https://github.com/Lundalogik/lime-elements/pull/2802/commits/892fe1625cd4620774e37ce7f70bdef11d3d9a24